### PR TITLE
initial commit with skeleton implementation of functionality

### PR DESCRIPTION
**Pull request by svenngjeto**

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,303 @@
+.gradle
```
Comment (brinxmat) [inline .gitattributes:1]: Not sure this is right.

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    public void testDefaultConstructor() {
+        SearchResourcesApiHandler searchResourcesApiHandler = new SearchResourcesApiHandler();
+        assertNotNull(searchResourcesApiHandler);
+    }
+
+
+    @Test
+    @DisplayName("testConstructorWithParameters")
+    public void testConstructorWithParameterEnvironmentDefined() {
+        Environment environment = mock(Environment.class);
```
Comment (brinxmat) [inline search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java:31]: Move following to @Before setup

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    @Test
+    @DisplayName("testDefaultConstructor")
+    public void testDefaultConstructor() {
+        SearchResourcesApiHandler searchResourcesApiHandler = new SearchResourcesApiHandler();
+        assertNotNull(searchResourcesApiHandler);
+    }
+
+
+    @Test
+    @DisplayName("testConstructorWithParameters")
```
Comment (brinxmat) [inline search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java:29]: Test isn't necessary anymore as it is tested in every test

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+        SearchResourcesApiHandler searchResourcesApiHandler = new SearchResourcesApiHandler(environment);
+        assertNotNull(searchResourcesApiHandler);
+    }
+
+    @Test
+    @DisplayName("testProcessEmptyInputFails")
+    public void testProcessEmptyInputReturnsNull() throws ApiGatewayException {
+        Environment environment = mock(Environment.class);
+        SearchResourcesApiHandler searchResourcesApiHandler = new SearchResourcesApiHandler(environment);
+        assertNotNull(searchResourcesApiHandler);
```
Comment (brinxmat) [inline search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java:41]: We don't need to assert not null here, it is implicit

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        SearchResourcesRequest request = mock(SearchResourcesRequest.class);
+        SearchResourcesResponse response =  new SearchResourcesResponse();
+        Integer statusCode = searchResourcesApiHandler.getSuccessStatusCode(request, response);
+        assertNull(statusCode);
+    }
+
+    @Test
+    @DisplayName("testCreatingEmptySearchRequest")
+    public void testCreatingEmptySearchRequest() {
```
Comment (brinxmat) [inline search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java:62]: I think this test needs to call searchResourcesApiHandler, if that is the aim, or remove

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+        SearchResourcesRequest searchResourcesRequest = new SearchResourcesRequest();
+        assertNotNull(searchResourcesRequest);
+    }
+
+    @Test
+    @DisplayName("testCreatingEmptySearchResponse")
+    public void testCreatingEmptySearchResponse() {
```
Comment (brinxmat) [inline search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java:69]: Is this an implicit result of calling searchResourcesApiHandler with an empty SearchRequest?

File: template.yaml [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+  NvaSearchApi:
+    Type: AWS::Serverless::Api
+    Properties:
+      AccessLogSetting:
+        DestinationArn: !GetAtt ApiAccessLogGroup.Arn
+        Format: '{ "apiId": "$context.apiId", "requestId": "$context.requestId", "requestTime": "$context.requestTime", "requestTimeEpoch": "$context.requestTimeEpoch", "httpMethod": "$context.httpMethod", "path": "$context.path", "status": "$context.status",  "error.message": "$context.error.message" }'
+      StageName: Prod
+      EndpointConfiguration: REGIONAL
+      DefinitionBody:
+        swagger: '2.0'
```
Comment (brinxmat) [inline template.yaml:41]: We should use openapi 3

File: template.yaml [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+                - in: query
+                  name: query
+                  type: string
+                  required: true
+                  description: term to search for in resource.
+                - in: query
+                  name: results
+                  type: integer
+                  required: false
+                  description: Maximum number of wanted publications in response.
```
Comment (brinxmat) [inline template.yaml:63]: remove "wanted "

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: template.yaml (hunk 8 above)
Comment (svenngjeto) [reply template.yaml:63]: easy...

File: template.yaml (hunk 7 above)
Comment (svenngjeto) [reply template.yaml:41]: Parameters defined in openapi 3.0.3

File: .gitattributes (hunk 1 above)
Comment (svenngjeto) [reply .gitattributes:1]: Removed file, seems to be OK

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java (hunk 2 above)
Comment (svenngjeto) [reply search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java:31]: Moved initialization of tests to a beforeEach method

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java (hunk 3 above)
Comment (svenngjeto) [reply search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java:29]: Simplified tests

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java (hunk 4 above)
Comment (svenngjeto) [reply search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java:41]: remved

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java (hunk 5 above)
Comment (svenngjeto) [reply search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java:62]: Moved to its own test class to make code-coverage happy - will be filled with more logic when appropiate

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java (hunk 6 above)
Comment (svenngjeto) [reply search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java:69]: same, same as above

**Review by brinxmat — CHANGES_REQUESTED** [review]

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    @Test
+    @DisplayName("testDefaultConstructor")
+    public void testDefaultConstructor() {
+        SearchResourcesApiHandler searchResourcesApiHandler = new SearchResourcesApiHandler();
+        assertNotNull(searchResourcesApiHandler);
+    }
+
+    @Test
+    @DisplayName("testProcessEmptyInputFails")
+    public void testProcessEmptyInputReturnsNull() throws ApiGatewayException {
```
[the remark this answers is not in the document]
Comment (svenngjeto) [reply search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java]: Tests renamed to something useful - idiot-tests remoed as line coverage is still OK

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+    @Test
+    @DisplayName("testDefaultConstructor")
+    public void testDefaultConstructor() {
+        SearchResourcesApiHandler searchResourcesApiHandler = new SearchResourcesApiHandler();
+        assertNotNull(searchResourcesApiHandler);
+    }
+
+    @Test
+    @DisplayName("testProcessEmptyInputFails")
```
[the remark this answers is not in the document]
Comment (svenngjeto) [reply search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java]: What names... :-)  No need to repeat the obvious

File: README.md [hunk 11]
```diff
@@ -1,2 +1,15 @@
 # nva-search-api
 A common search API for NVA across resources
+
+
+Structure wanted in response from the search API
+```JSON
+{
+"createdDate": "2020-08-10T07:56:27.232208Z",
```
Comment (axthosarouris) [inline README.md:8]: better to create a resource file that is updated by a test and refer to it here.

Comment (svenngjeto) [reply README.md]: Seems to be no usable way to include files in a markdow, but has added a sample file to be used in tests later on...

File: search-resources-api/src/main/java/no/unit/nva/search/SearchResourcesApiHandler.java [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+import com.amazonaws.services.lambda.runtime.Context;
+import no.unit.nva.search.exception.InputException;
+import nva.commons.exceptions.ApiGatewayException;
+import nva.commons.handlers.ApiGatewayHandler;
+import nva.commons.handlers.RequestInfo;
+import nva.commons.handlers.RestRequestHandler;
+import nva.commons.utils.Environment;
+import org.slf4j.LoggerFactory;
+
+public class SearchResourcesApiHandler extends ApiGatewayHandler<SearchResourcesRequest, SearchResourcesResponse> {
```
Comment (janbj) [inline search-resources-api/src/main/java/no/unit/nva/search/SearchResourcesApiHandler.java:12]: Terminology question, what is a Resource? Is it a Supertype for Publication?

File: search-resources-api/src/main/java/no/unit/nva/search/SearchResourcesApiHandler.java [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+    /**
+     * Define the success status code.
+     *
+     * @param input  The request input.
+     * @param output The response output
+     * @return the success status code.
+     */
+    @Override
+    protected Integer getSuccessStatusCode(SearchResourcesRequest input, SearchResourcesResponse output) {
+        return null;
```
Comment (janbj) [inline search-resources-api/src/main/java/no/unit/nva/search/SearchResourcesApiHandler.java:52]: 200 OK?

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+    @Test
+    public void processInputReturnsNullWhenInputIsEmpty() throws ApiGatewayException {
+        SearchResourcesRequest input = mock(SearchResourcesRequest.class);
+        RequestInfo requestInfo = mock(RequestInfo.class);
+        Context context = mock(Context.class);
+        assertThrows(InputException.class, () ->  searchResourcesApiHandler.processInput(input, requestInfo, context));
+    }
+
+    @Test
+    public void testGetSuccessStatusCode() {
```
Comment (janbj) [inline search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java:42]: getSuccessStatusCodeReturnsNull

**Review by janbj — APPROVED** [review]

File: search-resources-api/src/main/java/no/unit/nva/search/SearchResourcesApiHandler.java (hunk 12 above)
Comment (svenngjeto) [reply search-resources-api/src/main/java/no/unit/nva/search/SearchResourcesApiHandler.java:12]: Something between... As for now it is the name of the dynamoDB table - but i could comain lots of resources...

File: search-resources-api/src/main/java/no/unit/nva/search/SearchResourcesApiHandler.java (hunk 13 above)
Comment (svenngjeto) [reply search-resources-api/src/main/java/no/unit/nva/search/SearchResourcesApiHandler.java]: 200 OK it is

File: search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java (hunk 14 above)
Comment (svenngjeto) [reply search-resources-api/src/test/java/no/unit/nva/search/SearchResourcesApiHandlerTest.java]: testing 200 ok

File: README.md [hunk 15]
```diff
@@ -3,13 +3,26 @@ A common search API for NVA across resources
...
-"owner": "sase@unit.no",
-"status": "Published"
-}
-```
\ No newline at end of file
+  "publishedDate":"2020-08-26",
+  "owner":"na@unit.no",
+  "contributor":[
+    {
+      "name":"Doe, John"
```
Comment (brinxmat) [inline README.md:14]: We probably want an id on the contributors

File: search-resources-api/src/test/resources/sample_data_in_response.json [hunk 16]
```diff
@@ -0,0 +1,20 @@
+{
+  "publishedDate":"2020-08-26",
+  "owner":"na@unit.no",
+  "contributor":[
+    {
+      "name":"Doe, John"
```
Comment (brinxmat) [inline search-resources-api/src/test/resources/sample_data_in_response.json:6]: We probably want an id on the contributors

**Review by brinxmat — APPROVED** [review]

**Pull request merged**